Update after 10 years

berndporr opened the pull request:
Awesome library. I've debugged it that it works cross platform with the native opencl drivers for Intel but also the openCL drivers for nvidia. A few things have changed in openCL so I've updated it that it compiles again. Changed the dir structure a bit to have separate subdirs for the demo, test and the library itself. Also creates a static library file now.
My plan is to use it for my deep neuronal filter which does forward and backward pass at every timestep so that library is ideal for it. It will probably diverge quite a lot from now so better I send you a PR which is still your library with same functionality.

berndporr committed:
Fixed compilation errors but bombs during test.

berndporr committed:
Added missing cmake directives but still bombs in the test-suite.

berndporr committed:
Fixed uninit variable.

berndporr committed:
Compiles and all tests passed

berndporr committed:
Fetching https instead.

berndporr committed:
Updated cmakelists more to recommended settings

berndporr committed:
Weedeed out conversion problems between size_t and uint.

berndporr committed:
Fixed all uint32 issues and a different call format for local buffers.

berndporr committed:
Restructured code so that we have a proper library

berndporr committed:
Removed temp files

berndporr committed:
Updated the build instructions according to the new dir structure

berndporr committed:
Got rid of c++20 warnings.

berndporr committed:
Added proper debug built

berndporr committed:
Needs to run a few times in the test suite

berndporr committed:
Longer test times

saelo commented:
Hi and really sorry for the super slow reply! Great to hear that you've found this interesting/useful, and that you've made it work properly!
I don't think I'll actually be able to maintain this code, or review patches. Would you also be happy with forking this repository (I guess you already have)? In that case I would just mark this repository as archived and update the readme to point to your fork. What do you think?
Cheers!

berndporr commented:
No worries. Feel free to link to my repo and archive yours. It's become very topical now with quite a few Intel graphics chips on RaspberryPI like SBCs. I like your approach as all the training is done in one method basically which allows me to turn that into online training as well.

berndporr committed:
Got rid of a compiler warning when copying an initialiser list on a vector

berndporr committed:
Added error margins for the tensor comparison.

The reason I've added is because the convolution and correlations
failed. They have of course a higher error because it accumulates.

berndporr committed:
The convolution / correlation unit tests fail

Both throw up large errors in some matrix elements. That
might be an accumulation but probably rather a real issue.
Needs to be investigated.

berndporr committed:
Tweaked a bit the threshold when two tensors are different!

berndporr committed:
Tweaked the unit test more to have a sensible runtime.

berndporr committed:
Updated the dataset pointer

berndporr committed:
Added absolute error margin back in

berndporr committed:
Prevent printing out empty openCL log messages

saelo commented:
Done: https://github.com/saelo/deeplearn/commit/abd356107027fd0d3f711e9a08a3e75f2f7eaacc :)
